app/js/deleteCategory.ts

```diff
@@ -0,0 +1,9 @@
+/**
+ * deletes category from database
+ * @param {number} category id
+ * @returns {Promise<*>} response from api
+ */
+async function deleteCategory(id: number) {
+    return await sendData(undefined, `/category/delete/${id}`);
```

Comment:
You only need to do `return await` inside a try/catch block. Can simply return (without await) here. 

See this to understand why: https://jakearchibald.com/2017/await-vs-return-vs-return-await/